File: DESIGN-MOD/flyweight.js
//TV
function TV(brand, size){
    this.brand = brand;
    this.size = size;
}

//TV工厂
var TVFactory = (function(){
    var createdTV = {};

    return {
        createTV: function(brand, size){
            var tv,
                key = brand + size;
            if(typeof createdTV[key] !== 'undefined'){
                return createdTV[key];
            }else{
                tv = new TV(brand, size);
                createdTV[key] = tv;
                return tv;
            }
        }
    }
})();

//TV管理器
function TVManager(){
    this.tvDatabase = {};
}
TVManager.prototype.addNewTV = function(owner, brand, size){
    tv = TVFactory.createTV(brand, size);
    this.tvDatabase[owner] = tv;
};
TVManager.prototype.showTV = function(){
    for(var item in this.tvDatabase){
        console.log(item + ': ' + this.tvDatabase[item].brand + ', ' + this.tvDatabase[item].size);
    }
};


var tvManager = new TVManager();
//同一品牌、同一尺寸的TV只会共享一个实例
tvManager.addNewTV('adult', 'TCL', 30);
tvManager.addNewTV('children', 'TCL', 30);

tvManager.showTV();

/*console out

adult: TCL, 30
children: TCL, 30
 */

File: DESIGN-MOD/flyweight.py
#-*- coding:utf-8 -*-

#TV
class TV(object):
    def __init__(self, brand, size):
        self.brand = brand
        self.size = str(size)

#TV工厂
class TVFactory(object):
    createdTV = {}

    @staticmethod
    def createTV(brand, size):
        tvStyle = brand+str(size)
        if TVFactory.createdTV.has_key(tvStyle):
            return TVFactory.createdTV[tvStyle]
        else:
            tv = TV(brand, size)
            TVFactory.createdTV[tvStyle] = tv
            return tv

#TV管理器
class TVManager(object):
    tvDatabase = {}

    def addNewTV(self, owner, brand, size):
        tv = TVFactory.createTV(brand, size)
        self.__class__.tvDatabase[owner] = tv

    def showTV(self):
        for item in self.__class__.tvDatabase:
            print item + ": " + self.__class__.tvDatabase[item].brand + ", " + self.__class__.tvDatabase[item].size

if __name__ == "__main__":

    tvManager = TVManager()
    #同一品牌、同一尺寸的TV只会共享一个实例
    tvManager.addNewTV("adult", "TCL", 30)
    tvManager.addNewTV("children", "TCL", 30)
    tvManager.showTV()

"""print out

adult: TCL, 30
children: TCL, 30
"""